homeassistant/components/tplink/config_flow.py

```diff
@@ -384,6 +424,85 @@ async def async_step_user_auth_confirm(
             description_placeholders=placeholders,
         )
 
+    async def async_step_camera_auth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Dialog that gives the user option to set camera credentials."""
+        errors: dict[str, str] = {}
+        device = self._discovered_device
+        assert device
+        camera_module = device.modules[Module.Camera]
+        if TYPE_CHECKING:
+            # self.host is set by async_step_user and async_step_pick_device
+            assert self.host is not None
+
+        if user_input:
+            live_view = user_input[CONF_LIVE_VIEW]
+            if not live_view:
+                return self._async_create_or_update_entry_from_device(
+                    device, camera_data={CONF_LIVE_VIEW: False}
+                )
+            un = user_input.get(CONF_USERNAME)
+            pw = user_input.get(CONF_PASSWORD)
+            if bool(un) != bool(pw):
+                errors["base"] = "both_or_none"
+            else:
+                if un:
+                    entry_data = {
+                        CONF_LIVE_VIEW: True,
+                        CONF_CAMERA_CREDENTIALS: {CONF_USERNAME: un, CONF_PASSWORD: pw},
+                    }
+                    if TYPE_CHECKING:
+                        assert pw
+                    camera_creds = Credentials(un, pw)
+                else:
+                    entry_data = {CONF_LIVE_VIEW: True}
+                    camera_creds = None
+                rtsp_url = camera_module.stream_rtsp_url(camera_creds)
+                assert rtsp_url
+                img = await ffmpeg.async_get_image(self.hass, rtsp_url)
+
+                if img:
+                    return self._async_create_or_update_entry_from_device(
+                        device, camera_data=entry_data
+                    )
+
+                if await async_has_stream_auth_error(self.hass, rtsp_url):
+                    errors["base"] = "invalid_camera_auth"
+                else:
+                    errors["base"] = "cannot_connect_camera"
```

Comment:
Like above, would it be possible to bail out early? Having so many conditionals in this method makes it rather hard to follow the code flow.

Reply:
As above can't really bail out but nesting is reduced and the checking of the stream moved into a separate method.